common/flatpak-ref.c

```diff
@@ -340,6 +340,32 @@ flatpak_ref_format_ref (FlatpakRef *self)
                                       priv->arch);
 }
 
+/**
+ * flatpak_ref_format_ref_cached:
+ * @self: a #FlatpakRef
+ *
+ * Like flatpak_ref_format_ref() but this returns the same string each time
+ * it's called rather than allocating a new one.
+ *
+ * Returns: (transfer none): string representation
+ *
+ * Since: 1.9.1
+ */
+const char *
+flatpak_ref_format_ref_cached (FlatpakRef *self)
+{
+  char *full_ref;
+
+  full_ref = g_object_get_data (G_OBJECT (self), "cached-full-ref");
```

Comment:
Its a bit weird to use GObject data here. Why not just add a cached_full_ref to FlatpakRefPrivate? Its not a public struct anyway.